pkg/controller/linstornodeset/linstornodeset_controller.go

```diff
@@ -156,43 +158,14 @@ func (r *ReconcileLinstorNodeSet) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
-	if pns.Spec.DrbdRepoCred == "" {
-		return reconcile.Result{}, fmt.Errorf("NS Reconcile: missing required parameter drbdRepoCred: outdated schema")
-	}
-
-	if pns.Spec.SatelliteImage == "" {
-		return reconcile.Result{}, fmt.Errorf("NS Reconcile: missing required parameter satelliteImage: outdated schema")
-	}
-
-	if pns.Spec.KernelModImage == "" {
-		return reconcile.Result{}, fmt.Errorf("NS Reconcile: missing required parameter kernelModImage: outdated schema")
-	}
-
-	log := logrus.WithFields(logrus.Fields{
-		"resquestName":      request.Name,
-		"resquestNamespace": request.Namespace,
-	})
-	log.Info("NS Reconcile: reconciling LinstorNodeSet")
+	log.Debug("check if all required fields are filled")
 
-	logrus.WithFields(logrus.Fields{
-		"name":      pns.Name,
-		"namespace": pns.Namespace,
-		"spec":      fmt.Sprintf("%+v", pns.Spec),
-	}).Debug("NS Reconcile: found LinstorNodeSet")
-
-	if pns.Status.SatelliteStatuses == nil {
-		pns.Status.SatelliteStatuses = make([]*piraeusv1alpha1.SatelliteStatus, 0)
```

Comment:
Why can we remove these lines? We need to be robust against status serialization errors due to nil lists.

Reply:
The only point where we should update the status field is in the `reconcileStatus` method. Otherwise there is no guarantee that the spec will be saved/updated.

I made some changes to `reconcileStatus()` that ensure that all lists are initialized as empty.
This should be enough, as an "empty" status will not be serialized at all, and if it is set, it will be by our `reconcileStatus()`